components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -14588,69 +14588,86 @@ public void addBotDataEmailConfiguration(String email) throws SQLException, APIM
+        String query = SQLConstants.BotDataConstants.GET_SAVED_ALERT_EMAILS;
+        try (Connection connection = APIMgtDBUtil.getConnection();
+             PreparedStatement ps = connection.prepareStatement(query)) {
+            ResultSet resultSet = ps.executeQuery();
             while (resultSet.next()) {
                 BotDetectionData botDetectedData = new BotDetectionData();
                 botDetectedData.setUuid(resultSet.getString("UUID"));
                 botDetectedData.setEmail(resultSet.getString("SUBSCRIBER_ADDRESS"));
                 list.add(botDetectedData);
             }
         } catch (SQLException e) {
-            handleException("Failed to retrieve saved email types by tenant Name. ", e);
-        } finally {
-            APIMgtDBUtil.closeAllConnections(ps, conn, resultSet);
+            handleException("Error while retrieving bot detection alert subscriptions", e);
         }
         return list;
-
     }
 
     /**
-     * Delete email list from the database by using the tenantDomain
+     * Delete a bot detection alert subscription
+     *
+     * @param uuid uuid of the subscription
+     * @throws APIManagementException if an error occurs when deleting a bot detection alert subscription
      */
-    public void deleteBotDataEmailList(String uuid) throws APIManagementException, SQLException {
+    public void deleteBotDetectionAlertSubscription(String uuid) throws APIManagementException {
 
-        Connection connection;
-        PreparedStatement ps = null;
-        ResultSet rs = null;
-        connection = APIMgtDBUtil.getConnection();
-        connection.setAutoCommit(false);
-
-        try {
+        String query = SQLConstants.BotDataConstants.DELETE_EMAIL_BY_UUID;
+        try (Connection connection = APIMgtDBUtil.getConnection();
+             PreparedStatement ps = connection.prepareStatement(query)) {
             connection.setAutoCommit(false);
-            String deleteEmail = SQLConstants.BotDataConstants.DELETE_EMAIL_BY_UUID;
-            ps = connection.prepareStatement(deleteEmail);
             ps.setString(1, uuid);
             ps.execute();
             connection.commit();
         } catch (SQLException e) {
-            connection.rollback();
-            handleException("Failed to delete alert email data.", e);
-        } finally {
-            APIMgtDBUtil.closeAllConnections(ps, connection, rs);
+            handleException("Error while deleting bot detection alert subscription", e);
+        }
+    }
+
+    /**
+     * Retrieve a bot detection alert subscription by querying a particular field (uuid or email)
+     *
+     * @param field field to be queried to obtain the bot detection alert subscription. Can be uuid or email
+     * @param value value corresponding to the field (uuid or email value)
+     * @return if subscription exist, returns the bot detection alert subscription, else returns a null object
+     * @throws APIManagementException
```

Comment:
Add description for throws

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/9702/commits/f39ed1e5333a9182f3bae96a5b30ec0eee6bcbc9